capstone/src/main/java/com/google/univiz/api/VisResource.java

```diff
@@ -1,7 +1,40 @@
 package com.google.univiz.api;
 
+import com.google.univiz.CollegeData;
+import com.google.univiz.api.dataviz.Deadline;
+import com.google.univiz.api.dataviz.Timeline;
+import com.google.univiz.api.dataviz.TuitionStats;
+import java.util.List;
+
 /**
  * The VisResource interface is responsible for all data management/processing specific for the vis
  * feature of the UniViz webapp.
  */
-public interface VisResource {}
+public interface VisResource {
+
+  /**
+   * For every college the user selected, obtain the raw statistics and compute summary statistics
+   * on tuition-related data.
+   *
+   * @param colleges the raw data for each of the user's selected colleges
+   * @return the list of statistics per college.
+   */
+  List<TuitionStats> getTuitionStats(List<CollegeData> colleges);
+
+  /**
+   * For every college the user selected, obtain the recommended set of actions for the user to take
+   * and the recommended time range for each action.
+   *
+   * @param colleges the raw data for each of the user's selected colleges
+   * @return the actions/time range pairs for each college.
+   */
+  List<Timeline> getRecommendedTimeline(List<CollegeData> colleges);
```

Comment:
What's the difference between deadline and timeline? They seem very similar. Is timeline a collection of deadlines?

Reply:
A timeline ideally would have a range of deadlines associated with the various steps of completing an application. For instance an example timeline could have all of the following included:
- contacting potential application references for the first three weeks of the application period.
- writing the essay for the next three weeks.
- researching in depth into the university for the next two weeks.
whereas a deadline type would consist of application start/end dates only.